### DISCUSSION TITLE
How to add/create custom tab in GitHub repo using GitHub api?

### DISCUSSION THREAD

**Discussion by sububiker** [community/community#64138] [Apps, API and Webhooks]
Please share the feedback

**Comment by nirav-gajera** [comment]
To add/create a custom tab in a GitHub repository using the GitHub API, you can use the following steps:

1. **Make sure that you have the correct permissions.** You need to have the `repo` scope to be able to add/create a custom tab.
2. **Choose a name for your tab.** The name of your tab must be unique within your repository.
3. **Choose a content type for your tab.** The content type of your tab can be either `markdown` or `html`.
4. **Create a JSON payload that contains the name and content type of your tab.** The JSON payload should be in the following format:

```
{
  "name": "Your tab name",
  "content_type": "markdown"
}
```

5. **Make a POST request to the following endpoint:**

```
https://api.github.com/repos/OWNER/REPO/contents/tabs/YOUR_TAB_NAME
```

Replace `OWNER` with the name of the owner of the repository, `REPO` with the name of the repository, and `YOUR_TAB_NAME` with the name of your tab.

6. The response to the POST request will be a JSON object that contains the ID of your tab.

Here is an example of a POST request that you can use to add/create a custom tab:

```
curl -X POST \
  -H "Authorization: bearer YOUR_ACCESS_TOKEN" \
  -H "Content-Type: application/json" \
  -d '{
    "name": "Your tab name",
    "content_type": "markdown"
  }' \
  https://api.github.com/repos/OWNER/REPO/contents/tabs/YOUR_TAB_NAME
```

Replace `YOUR_ACCESS_TOKEN` with your GitHub access token.

Once you have added/created a custom tab, you can view it in the GitHub repository by clicking on the `Tabs` tab.
 -if you find it useful then mark as and answer please

**Reply by alexec** [reply]
This does not appear to work for my GitHub.

**Reply by bitti** [reply]
What is this? Just some copy & pasted chat GTP answer?

**Comment by amartyaroy** [comment]
Hi @nirav-gajera 
Is this solution applicable to only github.com users? For GHES user, any idea how we can do that from the REST api's?

**Reply by nirav-gajera** [reply]
same for GHES as well

**Reply by amartyaroy** [reply]
I tried this api but it was giving me 404 error

**Reply by amartyaroy** [reply]
In the actual documentation also i didnt find anything about such api's. Can you provide me the link if you have any that will be great help

**Reply by 13steinj** [reply]
[The /contents provides the repository file contents API...](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#get-repository-content)

I can't find a way to do this with the official API :(

**Reply by marcellodesales** [reply]
I can certify that it is indeed missing from GHE 3.10 version... 

* 404 using the Content API 
* 406 when using the hack of using the form-based API used in the UI 

![Screenshot 2023-12-18 at 12 32 51 PM](https://github.com/community/community/assets/131457/40c220ca-1f38-4b46-a46e-39e93304b266)
